src/main/java/com/mapr/synth/Synth.java

```diff
@@ -248,10 +269,20 @@ public static void header(Format format, List<String> names, PrintStream out) th
             }
         }
 
-        public static int generateFile(Options opts, SchemaSampler s, PrintStream out, int count) throws IOException {
-            for (int i = 0; i < count; i++) {
-                format(opts.format, opts.quote, s.getFieldNames(), s.sample(), out);
+
+        public static int generateFile(Options opts, SchemaSampler s, Template template, PrintStream out, int count) throws IOException {
+            if (template != null) {
+                PrintWriter writer = new PrintWriter(out);
```

Comment:
The `out` PrintStream is opened elsewhere and it would be 'unpolite' to close it here. In case of multiple files (when -output is specified) the PrintStreams are already wrapped with a try-with-resources statement (on ReportingWorker#call() method). On the other case the PrintStream is System.out that I believe cannot be reopened once the resources are freed after close() call.

PrintWriter here acts as wrapper for PrintStream because Freemarker requires a Writer instead of a Stream. Closing the writer would also close the underlying stream.